server/controllers/AutomationController.js

```diff
@@ -0,0 +1,283 @@
+
+    if (
+        phase === 'RD' ||
+        (phase === 'DRAFT' && testPlanRun.testPlanReport.markedFinalAt === null)
+    ) {
+        return null;
+    }
+
+    const { testPlanReport } = await populateData({
+        testPlanReportId: testPlanRun.testPlanReport.id
+    });
+
+    return await getFinalizedTestResults({
+        ...testPlanReport
+    });
+};
+
+const updateOrCreateTestResultWithResponses = async ({
+    testCsvRow,
+    testPlanRun,
+    responses,
+    atVersionId,
+    browserVersionId
+}) => {
+    const allTestsForTestPlanVersion = await testsResolver(
+        testPlanRun.testPlanReport.testPlanVersion
+    );
+    const testId = allTestsForTestPlanVersion.find(
+        test => parseInt(test.rowNumber, 10) === testCsvRow
+    )?.id;
+
+    if (testId === undefined) {
+        throwNoTestFoundError(testCsvRow);
+    }
+
+    const { testResult } = await findOrCreateTestResult({
+        testId,
+        testPlanRunId: testPlanRun.id,
+        atVersionId,
+        browserVersionId
+    });
+
+    const historicalTestResults = await getApprovedFinalizedTestResults(
+        testPlanRun
+    );
+
+    const historicalTestResult = historicalTestResults?.find(each => {
+        return each.testId === testId;
+    });
+
+    if (
+        historicalTestResult &&
+        historicalTestResult.scenarioResults?.length !==
+            testResult.scenarioResults.length
+    ) {
+        throw new Error(
+            'Historical test result does not match current test result'
+        );
+    }
+
+    const getAutomatedResultFromOutput = ({ baseTestResult, outputs }) => ({
+        ...baseTestResult,
+        atVersionId,
+        browserVersionId,
+        scenarioResults: baseTestResult.scenarioResults.map(
+            (scenarioResult, i) => {
+                // Check if output matches historical output
+                const outputMatches =
+                    historicalTestResult &&
+                    historicalTestResult.scenarioResults[i] &&
+                    historicalTestResult.scenarioResults[i].output ===
+                        outputs[i];
+
+                return {
+                    ...scenarioResult,
+                    output: outputs[i],
+                    assertionResults: scenarioResult.assertionResults.map(
+                        (assertionResult, j) => ({
+                            ...assertionResult,
+                            passed: outputMatches
+                                ? historicalTestResult.scenarioResults[i]
+                                      .assertionResults[j].passed
+                                : false,
+                            failedReason: outputMatches
+                                ? historicalTestResult.scenarioResults[i]
+                                      .assertionResults[j].failedReason
+                                : 'AUTOMATED_OUTPUT'
+                        })
+                    ),
+                    unexpectedBehaviors: []
```

Comment:
I'm not sure if it's happening in this part of the code, but I noticed that in the automated results, under the "Were there additional undesirable behaviors?" question, the radio button reading "No, there were no additional undesirable behaviors" was preselected, even without a human deciding so.

It feels like the more desirable outcome would be to leave the checkbox unchecked.

I believe the app already supports this if you set the unexpectedBehaviors array above as null (or possibly undefined).

Reply:
I agree! Changed!